src/providers/seventv/SeventvEmotes.cpp

```diff
@@ -139,6 +145,57 @@ namespace {
 
         return emotes;
     }
+
+    void updateEmoteMapPtr(Atomic<std::shared_ptr<const EmoteMap>> &map,
+                           EmoteMap &&updatedMap)
+    {
+        map.set(std::make_shared<EmoteMap>(std::move(updatedMap)));
+    }
+
+    boost::optional<EmotePtr> findEmote(
+        const std::shared_ptr<const EmoteMap> &map,
+        const QString &emoteBaseName, const QJsonObject &emoteJson)
+    {
+        auto id = emoteJson["id"].toString();
+
+        // Step 1: check if the emote is added with the base name
+        auto mapIt = map->find(EmoteName{emoteBaseName});
+        // We still need to check for the id!
+        if (mapIt != map->end() && mapIt->second->homePage.string.endsWith(id))
+        {
+            return mapIt->second;
+        }
+
+        std::lock_guard<std::mutex> guard(emoteCacheMutex);
+
+        // Step 2: check the cache for the emote
+        auto emote = emoteCache[EmoteId{id}].lock();
+        if (emote)
+        {
+            auto cacheIt = map->find(emote->name);
+            // Same as above, make sure it's actually the correct emote
+            if (cacheIt != map->end() &&
+                cacheIt->second->homePage.string.endsWith(id))
+            {
+                return cacheIt->second;
+            }
+        }
+        // Step 3: Since the emote is not added, and it's not even in the cache,
+        //         we need to check if the emote is added.
+        //         This is expensive but the cache entry may be corrupted
+        //         when an emote was added with a different alias in some other
+        //         channel.
+        for (auto it = map->begin(); it != map->end(); it++)
```

Comment:
Why not use a ranged for here? You can just dereference the shared_ptr to get a reference to the EmoteMap and then use it accordingly in the entire function